juniper_official/Kernel/detect-routing-socket-errors-iagent.rule

```diff
@@ -0,0 +1,125 @@
+/*
+ * This rule checks for the total number of routing socket errors at any given time in the system.
+ * Routing socket errors includes errors returned from rtsock, veto and mproc layers.
+ * It monitors the total number of routing socket errors and raises an alarm if there are too many of
+ * these detected for significant duration of time.
+ *
+ * This rule will check if the error has benn increased by 100 from its previous value in ALL the samples
+ * for the last 70 or 130 seconds and raises the alarm accordingly.
+ */
+iceberg {
+    topic kernel.rtsock {
+        rule detect-routing-socket-errors-iagent {
+            synopsis "Detect Routing Socket errors in the system.";
+            description "Detect errors in rtsock, veto and mproc layers.";
+            sensor rtsock-error-count-sensor {
+                synopsis "Rtsock error count sensor.";
+                description "sensor to keep track of rtsock error counts.";
+                iAgent {
+                    file rtsock-error-count.yml;
+                    table getRtsockErrorCountTable;
+                    frequency 20s;
```

Comment:
If the rule is not mission critical, please increase sensor frequency

Reply:
This is a critical parameter. Hence the frequency is 20s.